runtime/worker.rs

```diff
@@ -313,14 +313,14 @@ impl MainWorker {
     let startup_snapshot = options.startup_snapshot
       .expect("deno_runtime startup snapshot is not available with 'create_runtime_snapshot' Cargo feature.");
 
-    // Clear extension modules from the module map, except preserve `ext:deno_node`
-    // modules as `node:` specifiers.
-    let rename_modules = Some(
-      deno_node::SUPPORTED_BUILTIN_NODE_MODULES
-        .iter()
-        .map(|p| (p.ext_specifier, p.specifier))
-        .collect(),
-    );
+    // // Clear extension modules from the module map, except preserve `ext:deno_node`
+    // // modules as `node:` specifiers.
+    // let rename_modules = Some(
+    //   deno_node::SUPPORTED_BUILTIN_NODE_MODULES
+    //     .iter()
+    //     .map(|p| (p.ext_specifier, p.specifier))
+    //     .collect(),
+    // );
```

Comment:
The list of cleared module exceptions should be preserved, except without actually renaming them i.e. `[("node:assert", "node:assert"), ("node:buffer", "node:buffer"), ...]`. We can also remove the rename capability from runtime options so it just does that.

Reply:
Thanks for the pointer, but that doesn't really fix the build error (since this is runtime code). I believe the problem lies with the fact that I used `customizer` property that makes these files be defined multiple times when building a snapshot from snapshot. Any ideas?

Reply:
Ah, yeah using the customizer doesn't make the declarations predicated on whether or not `init_ops()` or `init_ops_with_esm()` is called...

Maybe we should add a separate `esm_customizer = ` which is only called in the `init_ops_and_esm()` case?

Reply:
So I had to restore back this list - to be honest, I'm not sure I fully understand why we have to do - why do we clear the module map here?

Reply:
Our way of allowing extension modules to import each other while hiding then from user code in the end is to clear the module map after runtime init. Or partially clear it in this case

Reply:
Got it, renamed to `preserve_snapshotted_modules` and made it accept `&'static [&'static str]` instead.

Reply:
> renamed to `preserve_snapshotted_modules`

That's better than `rename_modules`, but note the behaviour is to preserve all when it's passed as none. Also `extension_modules` might be more accurate. I suppose technically it's 'extension modules plus anything else which may have been snapshotted'.

Reply:
Good point, I'll rename it before landing - will have to move `core/` changes to https://github.com/denoland/deno_core

Reply:
I decided to keep the naming for now.